.github/workflows/lint-format.yml

```diff
@@ -0,0 +1,19 @@
+name: Lint & Format
+on: [pull_request]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: pip3 install ruff
+      - run: ruff check
+    env:
+      RUFF_OUTPUT_FORMAT: github
+  format:
```

Comment:
Does this format the code and push the formatted code to the branch?

Reply:
This only raises error if the code is not of correct format and has linting errors.